osu.Game/Screens/Select/Carousel/CarouselGroup.cs

```diff
@@ -36,10 +42,24 @@ public virtual void AddChild(CarouselItem i)
         {
             i.State.ValueChanged += state => ChildItemStateChanged(i, state.NewValue);
             i.ChildID = ++currentChildID;
-            InternalChildren.Add(i);
+
+            if (lastCriteria != null)
+            {
+                i.Filter(lastCriteria);
+
+                int index = InternalChildren.BinarySearch(i, criteriaComparer);
+                if (index < 0) index = ~index; // BinarySearch hacks multiple return values with 2's complement.
+
+                InternalChildren.Insert(index, i);
```

Comment:
> or, if there is no larger element, the bitwise complement of Count.

Seems like this should be clamped to prevent exceptions when the user doesn't have any beatmaps but does have a filter applied.

Reply:
Hmm, I copied this directly from `AddInPlace`..

Reply:
Nevermind, apparently insert does accept `index == Count`